### DISCUSSION TITLE
Any plans to update the IPU strategy API?

### DISCUSSION THREAD

**Discussion by daniellepintz** [Lightning-AI/pytorch-lightning#11743] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Jw if there are any plans to update the IPU strategy api. I ask because IPU is the only Strategy that uses these 9 hooks:

https://github.com/PyTorchLightning/pytorch-lightning/blob/8c07d8bf905e395bbd2142b5df7b185b8e936c41/pytorch_lightning/strategies/strategy.py#L446-L480

and it would be nice to simplify the Strategy API by deprecating these hooks.

**Comment by rohitgr7** [comment]
checking out some examples here: https://github.com/graphcore/poptorch/tree/sdk-release-2.4/examples,
they don't seem to use `attachToDevice` anywhere. I wonder why is that required, or is even required anymore. If not, the majority of them can be removed.

**Reply by daniellepintz** [reply]
Cool, that would be great! Any thoughts @SeanNaren?

**Reply by SeanNaren** [reply]
Since we can run inference/training in the same session we need to be able to attach the particular model to device. This is unfortunately still a requirement I think, till the IPU API updates(cc @davors72)

**Reply by ananthsub** [reply]
@SeanNaren any idea how this works in the MNIST example without the models explicitly being attached to the device? https://github.com/graphcore/poptorch/blob/ab184c575b52fd24f4621ef464f8db4e87a3cf3e/examples/mnist.py#L176-L180

**Reply by SeanNaren** [reply]
correct me if I'm wrong @davors72 but the above example will duplicate the model onto separate IPU devices?
